spec/moonshot/plugins/rotate_asg_instances/asg_spec.rb

```diff
@@ -108,64 +112,37 @@ def stub_cf_client
     end
   end
 
-  describe '#initialize' do
-    before(:each) do
-      allow(System).to receive(:new)
-        .and_return(system)
-      stub_cf_client
-    end
-    context 'when MOONSHOT_SSH_USER is not defined' do
-      it 'uses LOGNAME for ssh_user' do
-        ENV['LOGNAME'] = 'SemiCoolDude'
-        ENV.delete('MOONSHOT_SSH_USER')
-        expect(subject.instance_variable_get(:@ssh_user))
-          .to eq('SemiCoolDude')
-      end
-    end
-
-    context 'when MOONSHOT_SSH_USER is defined' do
-      it 'uses MOONSHOT_SSH_USER for ssh_user' do
-        ENV['MOONSHOT_SSH_USER'] = 'CoolDude'
-        expect(subject.instance_variable_get(:@ssh_user))
-          .to eq('CoolDude')
-      end
-    end
-  end
-
   describe '#shutdown_instance' do
-    let(:hostname) { 'ec2-54-236-102-14.compute-1.amazonaws.com' }
+    let(:public_ip_address) { '10.234.32.21' }
     let(:instance) { instance_double(Aws::EC2::Instance) }
+    let(:command_builder) { Moonshot::SSHCommandBuilder }
     subject { super().send(:shutdown_instance, instance_id) }
+    let(:set_ssh_options_to_nil) { moonshot_config.ssh_config.ssh_options = nil }
```

Comment:
Generally, the use of `let` should be declarative, not procedural.  In this case, it might look something like this:

```
let(:ssh_options) { nil }

before_each do
  moonshot_config.ssh_config.ssh_options = ssh_options
end

it 'issues a shutdown without options...' do
  ...
end

context 'when SSH options are defined'
  let(:ssh_optons) { '-v' }
  
  it 'issues a shutdown with options' do
    ...
  end
end
```

Reply:
I will correct it.

Reply:
I corrected. Please check.